WordPress/src/test/java/org/wordpress/android/ui/jetpack/scan/ThreatItemBuilderTest.kt

```diff
@@ -0,0 +1,236 @@
+        // Assert
+        assertThat(threatItem.header).isEqualTo(expectedHeader)
+    }
+
+    @Test
+    fun `builds threat sub header correctly for FileThreatModel`() {
+        // Arrange
+        val expectedSubHeader = UiStringResWithParams(
+            R.string.threat_item_sub_header_file_signature,
+            listOf(UiStringText(TEST_SIGNATURE))
+        )
+        // Act
+        val threatItem = buildThreatItem(fileThreatModel)
+        // Assert
+        assertThat(threatItem.subHeader).isEqualTo(expectedSubHeader)
+    }
+
+    @Test
+    fun `builds threat header correctly for VulnerableExtensionThreatModel of Plugin type`() {
+        // Arrange
+        val expectedHeader = UiStringResWithParams(
+            R.string.threat_item_header_vulnerable_plugin,
+            listOf(UiStringText(VULNERABLE_THREAT_SLUG), UiStringText(VULNERABLE_THREAT_VERSION))
+        )
+        // Act
+        val threatItem = buildThreatItem(vulnerableExtensionThreatModel)
+        // Assert
+        assertThat(threatItem.header).isEqualTo(expectedHeader)
+    }
+
+    @Test
+    fun `builds threat sub header correctly VulnerableExtensionThreatModel of Plugin type`() {
+        // Arrange
+        val expectedSubHeader = UiStringRes(R.string.threat_item_sub_header_vulnerable_plugin)
+        // Act
+        val threatItem = buildThreatItem(vulnerableExtensionThreatModel)
+        // Assert
+        assertThat(threatItem.subHeader).isEqualTo(expectedSubHeader)
+    }
+
+    @Test
+    fun `builds threat header correctly for VulnerableExtensionThreatModel of Theme type`() {
+        // Arrange
+        val expectedHeader = UiStringResWithParams(
+            R.string.threat_item_header_vulnerable_theme,
+            listOf(UiStringText(VULNERABLE_THREAT_SLUG), UiStringText(VULNERABLE_THREAT_VERSION))
+        )
+        // Act
+        val threatItem = buildThreatItem(vulnerableExtensionThreatModel.copy(extension = extension.copy(type = THEME)))
+        // Assert
+        assertThat(threatItem.header).isEqualTo(expectedHeader)
+    }
+
+    @Test
+    fun `builds threat sub header correctly VulnerableExtensionThreatModel of Theme type`() {
+        // Arrange
+        val expectedSubHeader = UiStringRes(R.string.threat_item_sub_header_vulnerable_theme)
+        // Act
+        val threatItem = buildThreatItem(vulnerableExtensionThreatModel.copy(extension = extension.copy(type = THEME)))
+        // Assert
+        assertThat(threatItem.subHeader).isEqualTo(expectedSubHeader)
+    }
+
+    @Test
+    fun `builds threat header correctly for GenericThreatModel`() {
+        // Arrange
+        val expectedHeader = UiStringRes(R.string.threat_item_header_threat_found)
+        // Act
+        val threatItem = buildThreatItem(genericThreatModel)
+        // Assert
+        assertThat(threatItem.header).isEqualTo(expectedHeader)
+    }
+
+    @Test
+    fun `builds threat sub header correctly Plugin GenericThreatModel`() {
```

Comment:
🔍 Nitpick: Missing "for" -> "correctly for VulnerableExtensionThreatModel"

Reply:
In c5d9fe2